File: src/main/java/com/indiepost/repository/CategoryRepository.java
package com.indiepost.repository;

import com.indiepost.model.Category;

import java.util.List;

/**
 * Created by jake on 7/26/16.
 */
public interface CategoryRepository {

    void save(Category category);

    void update(Category category);

    void delete(Category category);

    Category findById(Long id);

    Category findBySlug(String slug);

    List<Category> findByParentId(Long parentId);

    List<Category> findAll();
}

File: src/main/kotlin/com/indiepost/repository/CategoryRepository.kt
package com.indiepost.repository

import com.indiepost.model.Category

/**
 * Created by jake on 7/26/16.
 */
interface CategoryRepository {

    fun persist(category: Category)

    fun delete(category: Category)

    fun getReference(id: Long): Category?

    fun findById(id: Long): Category?

    fun findBySlug(slug: String): Category?

    fun findByParentId(parentId: Long): List<Category>

    fun findAll(): List<Category>
}
